components/org.wso2.carbon.identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -268,16 +253,67 @@ private static ResourceEndpointException buildResourceEndpointException(Response
      * @param resourceId resourceId of the resource which need to get deleted
      * @return validator obtain
      */
-    public boolean isResourceId(String resourceId) throws UMAClientException {
+    private static final boolean isResourceIdValid(String resourceId) throws UMAClientException {
 
         String validPattern = "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$";
         Pattern pattern = Pattern.compile(validPattern);
         Matcher match = pattern.matcher(resourceId);
-
         if (match.find()) {
             return true;
         } else {
-            return false;
+            log.error("Invalid ResourceId.");
```

Comment:
Why you are not returning false here,return false. the error should handle in the method above